src/System.IO.Compression/tests/ZipArchive/zip_CreateTests.cs

```diff
@@ -149,6 +149,88 @@ public static void CreateUncompressedArchive()
             }
         }
 
+        [Theory]
+        [InlineData("minimalTwoFiles")]
+        public static void CreateNormal_VerifyDataDescriptor(string folder)
+        {
+            void CreateEntry(ZipArchive archive, string fileName, string fileContents)
+            {
+                ZipArchiveEntry entry = archive.CreateEntry(fileName);
+                using (StreamWriter writer = new StreamWriter(entry.Open()))
+                {
+                    writer.Write(fileContents);
+                }
+            }
+
+            string fileName = folder + ".zip";
+
+            if (File.Exists(fileName))
+            {
+                File.Delete(fileName);
+            }
+
+            using (var fileStream = new FileStream(fileName, FileMode.Create))
+            {
+                // We need an unseekable stream so the data descriptor bit is turned on when saving
+                var wrappedStream = new WrappedStream(fileStream, true, true, false, null);
+
+                // Creation will go through the path that sets the data descriptor bit when the stream is unseekable
+                using (var archive = new ZipArchive(wrappedStream, ZipArchiveMode.Create))
+                {
+                    CreateEntry(archive, "A", "xxx");
+                    CreateEntry(archive, "B", "yyy");
+                }
+            }
+
+            using (var fileStream = new FileStream(fileName, FileMode.Open))
+            {
+                using (var memoryStream = new MemoryStream())
+                {
+                    fileStream.CopyTo(memoryStream);
+                    byte[] fileBytes = memoryStream.ToArray();
+
+                    // We expect the data descriptor to be on in the first local file header after creation
+                    Assert.Equal(8, fileBytes[6]);
+                    Assert.Equal(0, fileBytes[7]);
+                }
+            }
+
+            using (var fileStream = new FileStream(fileName, FileMode.Open))
+            {
+                // Update should flip the data descriptor bit to zero on save
+                using (var archive = new ZipArchive(fileStream, ZipArchiveMode.Update))
+                {
+                    ZipArchiveEntry entry = archive.Entries[0];
+                    using (var entryStream = entry.Open())
+                    {
+                        using (StreamWriter writer = new StreamWriter(entryStream))
+                        {
+                            writer.Write("zzz");
+                        }
+                    }
+                }
+            }
+
+            using (var fileStream = new FileStream(fileName, FileMode.Open))
```

Comment:
Nit, we usually skip braces/indent when nesting using statements directly eg
```c#
            using (var fileStream = new FileStream(fileName, FileMode.Open))
            using (var memoryStream = new MemoryStream())
            {
               ...
```

Reply:
Thanks. I'll simplify this.

Reply:
After addressing the suggestions, I won't be having two using statements one after the other, so I'll resolve this.